NetworkManager: update to 1.28.0.

codingHahn opened the pull request:
Builds for me.

Tests:
- [ ] x86
- [x] x86_64
- [ ] arm
- [ ] anything musl

Do we need to revbump some packages?

Closes #25891

fosslinux commented on srcpkgs/NetworkManager/template:
```diff
@@ -2,7 +2,7 @@
 # when updating to 1.24 or newer, drop the compat
 # patch in gnome-control-center and rebuild it
```
act on this?

ericonr commented:
@codingHahn ping?

codingHahn commented:
Sorry for the delay. x86 glibc fails during the tests. Any ideas?
Musl probably fails because of some patch. I will take a look.

pullmoll commented:
The current NetworkManager 1.22.10 no longer (cross) builds.
There is version 1.28.0 it seems, though.

ericonr commented:
Put tests behind `if [ "$XBPS_CHECK_PKGS" = full ]; then`

codingHahn commented:
> The current NetworkManager 1.22.10 no longer (cross) builds.
> There is version 1.28.0 it seems, though.

Should we update to 1.28 straight away?

ericonr commented:
> Should we update to 1.28 straight away?

Probably best, yes

codingHahn commented:
Updated to 1.28.0, dropped musl patch that did not apply cleanly (I did not find the patched sections in the files anymore). The check should now be behind `if [ "$XBPS_CHECK_PKGS" = full ]; then` . I hope I did that correctly. If you have a cleaner way, let me know.

codingHahn commented:
It seems that all -musl packages fail with this error:
```
[463/709] Compiling C object src/libNetworkManagerBase.a.p/NetworkManagerUtils.c.o
FAILED: src/libNetworkManagerBase.a.p/NetworkManagerUtils.c.o 
arm-linux-musleabihf-gcc -Isrc/libNetworkManagerBase.a.p -Isrc -I../src -Ilibnm-core -I../libnm-core -I. -I.. -Ishared -I../shared -I../shared/systemd/sd-adapt-shared -I../shared/systemd/src/basic -I../shared/systemd/src/shared -I/usr/arm-linux-musleabihf/usr/include/gio-unix-2.0 -I/usr/arm-linux-musleabihf/usr/include/libmount -I/usr/arm-linux-musleabihf/usr/include/blkid -I/usr/arm-linux-musleabihf/usr/include/glib-2.0 -I/usr/arm-linux-musleabihf/usr/lib/glib-2.0/include -I/usr/arm-linux-musleabihf/usr/include/elogind -I/usr/arm-linux-musleabihf/usr/include -flto -fdiagnostics-color=always -DNDEBUG -pipe -D_FILE_OFFSET_BITS=64 -std=gnu11 -fdata-sections -ffunction-sections -flto-partition=none -Wall -Wextra -Wdeclaration-after-statement -Wfloat-equal -Wformat-nonliteral -Wformat-security -Wimplicit-function-declaration -Winit-self -Wlogical-op -Wmissing-declarations -Wmissing-include-dirs -Wmissing-prototypes -Wpointer-arith -Wshadow -Wshift-negative-value -Wstrict-prototypes -Wundef -Wvla -Wno-duplicate-decl-specifier -Wno-format-truncation -Wno-format-y2k -Wno-missing-field-initializers -Wno-pragmas -Wno-sign-compare -Wno-unknown-pragmas -Wno-unused-parameter -Wpointer-arith -Wshadow -Wstrict-prototypes -Wundef -fno-strict-aliasing -Wimplicit-fallthrough -fstack-clash-protection -D_FORTIFY_SOURCE=2 -O2 -march=armv6 -mfpu=vfp -mfloat-abi=hard -DRTLD_DEEPBIND=0 -fPIC -pthread -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_40 -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_40 -DNETWORKMANAGER_COMPILATION=NM_NETWORKMANAGER_COMPILATION_DAEMON -MD -MQ src/libNetworkManagerBase.a.p/NetworkManagerUtils.c.o -MF src/libNetworkManagerBase.a.p/NetworkManagerUtils.c.o.d -o src/libNetworkManagerBase.a.p/NetworkManagerUtils.c.o -c ../src/NetworkManagerUtils.c
In file included from ../src/NetworkManagerUtils.c:12:
../src/NetworkManagerUtils.c: In function 'nm_utils_tfilters_from_tc_setting':
../src/NetworkManagerUtils.c:1222:51: error: 'ETH_P_ALL' undeclared (first use in this function)
 1222 |         tfilter->info        = TC_H_MAKE(0, htons(ETH_P_ALL));
      |                                                   ^~~~~~~~~
../src/NetworkManagerUtils.c:1222:51: note: each undeclared identifier is reported only once for each function it appears in
```
After a quick search it seems that this constant is either from the linux kernel or glibc.

And all non-musl ones fail here:
```
[456/841] Generating nm-settings-docs-gir.xml with a custom command
FAILED: libnm/nm-settings-docs-gir.xml 
env GI_TYPELIB_PATH=/builddir/NetworkManager-1.28.0/build/libnm LD_LIBRARY_PATH=/builddir/NetworkManager-1.28.0/build/libnm sh -c '/usr/bin/qemu-aarch64-static -L /usr/aarch64-linux-gnu /usr/aarch64-linux-gnu/bin/python3' /builddir/NetworkManager-1.28.0/tools/generate-docs-nm-settings-docs-gir.py --lib-path /builddir/NetworkManager-1.28.0/build/libnm --gir libnm/NM-1.0.gir --output libnm/nm-settings-docs-gir.xml
```
I'll have to look at the patch again that was supposed to fix that (fix-cross-gir.patch)

I will fix the lint error this weekend.

ericonr commented:
We should definitely have that definition, will have to check why it isn't found.

ericonr commented on srcpkgs/NetworkManager/template:
```diff
@@ -68,6 +67,16 @@ pre_configure() {
 
+do_check() {
+    if [ "$XBPS_CHECK_PKGS" = full ]; then
+        : ${make_cmd:=ninja}
+        : ${make_check_target:=test}
+        : ${meson_builddir:=build}
+
+        ${make_cmd} -C ${meson_builddir} ${makejobs} ${make_check_args} ${make_check_target}
```
```suggestion
        ninja -C build test
```

codingHahn replied:
Yeah, seems better. I just copied it from the buildstyle. Will change that

codingHahn commented:
Superseded by #29342
